src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -967,9 +985,17 @@ void give_task_to_local_scheduler_retry(UniqueID id,
   ActorID actor_id = TaskSpec_actor_id(spec);
   CHECK(state->actor_mapping.count(actor_id) == 1);
 
-  give_task_to_local_scheduler(
-      state, state->algorithm_state, *execution_spec,
-      state->actor_mapping[actor_id].local_scheduler_id);
+  if (DBClientID_equal(state->actor_mapping[actor_id].local_scheduler_id,
```

Comment:
This is an optimization to avoid going through Redis when the local scheduler should keep the task for itself, right? It does not affect correctness, right?

Reply:
Yup, it's an optimization. I think it would be nice to keep, since otherwise we get a bunch of spurious warning messages (`"Local scheduler is trying to assign a task to itself."`). You can see this if you increase the number of forks or tasks in the unit test that I added in `actor_test.py`.